open_seq2seq/data/speech2text/speech2text.py

```diff
@@ -192,6 +193,59 @@ def build_graph(self):
     else:
       self._input_tensors['source_ids'] = [x_id]
 
+  def create_interactive_placeholders(self):
+    self._x = tf.placeholder(dtype=self.params['dtype'])
+    self._x_length = tf.placeholder(dtype=tf.int32)
+    self._x_id = tf.placeholder(dtype=tf.int32)
+
```

Comment:
You can specify shape parameter directly when creating placeholder, just use shape=... . This will make this code a bit more compact and easier to read.